lib/top-bar.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux'
+
+import { setRevisionState } from './state/revisions/actions';
+import { toggleLayoutType } from './state/settings/actions'
+import NewNoteIcon from './icons/new-note';
+import NoteToolbar from './note-toolbar';
+
+class TopBar extends Component {
+
+	static propTypes = {
+		layoutType: PropTypes.string,
+		revisionState: PropTypes.object,
+		onToggle: PropTypes.func,
+		onRevisionChange: PropTypes.func,
+		note: PropTypes.object,
+		onNewNote: PropTypes.func,
+		onTrashNote: PropTypes.func,
+		onRestoreNote: PropTypes.func,
+		onShareNote: PropTypes.func,
+		onDeleteNoteForever: PropTypes.func,
+		onRevisions: PropTypes.func,
+		onCloseNote: PropTypes.func,
+		onNoteInfo: PropTypes.func,
+	}
+
+	toggleIcon() {
+		switch ( this.props.layoutType ) {
+			case '3-col' : return '3 COLUMNS';
+			case '2-col' : return '2 COLUMNS';
+			case '1-col' : return 'EDITOR';
+		}
```

Comment:
I kinda hate the naming I used there (and of course the returns are only placeholder for the actual icons), but at the same time I honestly feel the good old switch makes it more readable at a quick glance.

As for the extendibility: adding a new panel in the main layout is also a heavy design concern and I don't think it that adding a case would prove that hard in the off chance we need it.